markets/perps-market/contracts/storage/Position.sol

```diff
@@ -72,7 +84,31 @@ library Position {
 
         int priceShift = price.toInt() - self.latestInteractionPrice.toInt();
         pricePnl = self.size.mulDecimal(priceShift);
-        totalPnl = pricePnl + accruedFunding;
+
+        chargedInterest = interestAccrued(self, price);
+
+        totalPnl = pricePnl + accruedFunding - chargedInterest.toInt();
+    }
+
+    function interestAccrued(
+        Data storage self,
+        uint price
+    ) internal view returns (uint256 chargedInterest) {
+        uint nextInterestAccrued = InterestRate.load().calculateNextInterest();
```

Comment:
`calculateNextInterest` does this: `return self.interestAccrued + unrecordedInterest(self);` where `unrecordedInterest` return the raw value based on the stored interestRate. (`self.interestRate.mulDecimalUint128(proportionalElapsed(self)).to256();`)

Is there a reason this deviates from how we do unrecorded funding?
Inside unrecordedInterest I expected to see:
```
uint256 avgUtilizationRate =(self.interestRate + currentInterestRate(...)).divDecimal(DecimalMath.UNIT * 2)
return avgUtilizationRate.mulDecimalUint128(proportionalElapsed(self)).to256();
```
With the idea that the unrecorded funding is the average between the previous funding rate and the current one.

Basically the question comes down to:
If I open a position when utilisation is 1%
And close when it's 3%
With the average we would charge 2%. Whereas they way it's written here I think we would charge 3%.

Reply:
you get charged 3% from the time it changed to 3% to when the trader closed the position, _not_ the average.  

the trader does not get charged 3%, they get charged 1% on their position until the rate goes to 3% (due to another position or LP undelegating), then charged 3% from that time forward till they close.  you can't average it, its time based.

funding is using average for calculating rate across continuous time NOT accrued funding per unit of market.